gradle-plugins/src/main/java/com/linkedin/pegasus/gradle/PegasusPlugin.java

```diff
@@ -1693,6 +1646,34 @@ protected GenerateDataTemplateTask configureDataTemplateGeneration(Project proje
     prepareLegacySchemasForPublishTask.dependsOn(destroyStaleFiles);
     dataTemplateJarDepends.add(prepareLegacySchemasForPublishTask);
 
+    // extension schema directory
+    File extensionSchemaDir = project.file(getExtensionSchemaPath(project, sourceSet));
+
+    if (!SharedFileUtils.getSuffixedFiles(project, extensionSchemaDir, PDL_FILE_SUFFIX).isEmpty()) {
+      // Validate extension schemas if extension schemas are provided.
+      ValidateExtensionSchemaTask validateExtensionSchemaTask = project.getTasks()
+          .create(sourceSet.getTaskName("validate", "ExtensionSchemas"), ValidateExtensionSchemaTask.class, task -> {
+            task.setInputDir(extensionSchemaDir);
+            task.setResolverPath(
+                getDataModelConfig(project, sourceSet).plus(project.files(getDataSchemaPath(project, sourceSet))));
+            task.setClassPath(project.getConfigurations().getByName("pegasusPlugin"));
+            if (isPropertyTrue(project, ENABLE_ARG_FILE)) {
+              task.setEnableArgFile(true);
+            }
+          });
+
+      Task prepareExtensionSchemasForPublishTask = project.getTasks()
```

Comment:
Question, what is the previous task of task: `prepareExtensionSchemasForPublishTask` and what is the next task?
I feel `prepare` should be part of some task, not itself a task. But let me know if I miss understand something.

Reply:
I see `validateExtensionSchemaTask` and `copyPdscSchemasTask` is the previous task of this one.

Reply:
Discussed offline, not a question any more.